### PR TITLE
feat(charms-introducing-screen): UX & navigation

### PR DESCRIPTION
**Pull request by VladislavRiabko**
- refactor Carousel component
- add CharmsIntroducing screen

### CODE REVIEW

File: app/components/Carousel/Carousel.props.ts [hunk 1]
```diff
@@ -3,4 +3,5 @@ import { CarouselType, ICarouselItem } from './Carousel.data';
 export interface ICarouselProps {
   data: ICarouselItem[];
   preset: CarouselType;
+  setIndex?: (item: number) => void;
```
Comment (emberglaz) [inline app/components/Carousel/Carousel.props.ts]: `setCurrentPosition`

File: app/components/Carousel/Carousel.tsx [hunk 2]
```diff
@@ -15,12 +15,21 @@ import {
 } from './components';
 import { CarouselType, ICarouselItem } from './Carousel.data';
 
-export const Carousel: React.FC<ICarouselProps> = ({ preset, data }) => {
+export const Carousel: React.FC<ICarouselProps> = ({
+  preset,
+  data,
+  setIndex,
```
Comment (emberglaz) [inline app/components/Carousel/Carousel.tsx]: the same

File: app/screens/Onboarding/Parents/CharmsIntroducing/CharmsIntroducing.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          )}
+          onSubmit={_.noop}>
+          <MainHeader
+            leftIcon={IMAGES.WHITE_BACK_ARROW}
+            onLeftIconPress={navigation.goBack}
+          />
+          <Carousel
+            data={CHARMS_CAROUSEL}
+            preset={CarouselType.ImageTitleSubTitle}
+            setIndex={setIndex}
```
Comment (emberglaz) [inline app/screens/Onboarding/Parents/CharmsIntroducing/CharmsIntroducing.tsx]: the same

File: app/screens/Onboarding/Parents/CharmsIntroducing/CharmsIntroducing.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  ABSOLUTE_PROGRESS_VALUE,
+  CHARMS_CAROUSEL,
+} from './CharmsIntroducing.data';
+import { ICharmsIntroducingScreenProps } from './CharmsIntroducing.props';
+
+export const CharmsIntroducingScreen: React.FC<ICharmsIntroducingScreenProps> =
+  ({ navigation }) => {
+    const { t } = useTranslation();
+
+    const [index, setIndex] = useState<number>(0);
```
Comment (emberglaz) [inline app/screens/Onboarding/Parents/CharmsIntroducing/CharmsIntroducing.tsx]: the same

File: app/services/localization/languages/en.ts [hunk 5]
```diff
@@ -2,7 +2,7 @@ export const en = {
   buttons: {
     confirm: 'Confirm',
     enter: 'Enter',
-    next: 'Next',
+    next: 'NEXT',
```
Comment (emberglaz) [inline app/services/localization/languages/en.ts]: `next: 'Next'`

File: app/screens/Onboarding/Parents/CharmsIntroducing/CharmsIntroducing.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    return (
+      // TODO: remove noop to real function
+      <SafeAreaView style={[generalStyles.whFlex]}>
+        <BottomButtonView
+          buttonTitle={t(
+            index > ABSOLUTE_PROGRESS_VALUE ? 'buttons.ready' : 'buttons.next',
```
Comment (emberglaz) [inline app/screens/Onboarding/Parents/CharmsIntroducing/CharmsIntroducing.tsx]: `(index > ABSOLUTE_PROGRESS_VALUE ? 'buttons.ready' : 'buttons.next').toUpperCase()`

**Review by emberglaz — CHANGES_REQUESTED** [review]

File: app/components/Carousel/Carousel.props.ts [hunk 7]
```diff
@@ -3,4 +3,5 @@ import { CarouselType, ICarouselItem } from './Carousel.data';
 export interface ICarouselProps {
   data: ICarouselItem[];
   preset: CarouselType;
+  setCurrenPosition?: (item: number) => void;
```
Comment (Reykjavik151) [inline app/components/Carousel/Carousel.props.ts]: rename `setCurrentPossition`

File: app/components/Carousel/components/ImageSubTitle/ImageSubTitle.tsx [hunk 8]
```diff
@@ -1,13 +1,15 @@
...
 
 import { ExtendedText } from '../../../ExtendedText';
 import { IImageSubTitleProps } from './ImageSubTitle.props';
 
 export const ImageSubTitle: React.FC<IImageSubTitleProps> = ({ data }) => {
+  const { t } = useTranslation();
   return (
     <View>
-      <ExtendedText>{data.subTitle}</ExtendedText>
+      {data.subTitle && <ExtendedText>{t(data.subTitle)}</ExtendedText>}
```
Comment (Reykjavik151) [inline app/components/Carousel/components/ImageSubTitle/ImageSubTitle.tsx]: use `data.subTitleID` if you will translate this value

File: app/components/Carousel/components/ImageTitleSubTitle/ImageTitleSubTitle.tsx [hunk 9]
```diff
@@ -7,11 +8,12 @@ import { IImageTitleSubTitleProps } from './ImageTitleSubTitle.props';
...
   data,
 }) => {
+  const { t } = useTranslation();
   return (
     <View>
       {data.image && <Image source={data.image} />}
-      <ExtendedText>{data.title}</ExtendedText>
-      <ExtendedText>{data.subTitle}</ExtendedText>
+      {data.title && <ExtendedText>{t(data.title)}</ExtendedText>}
+      {data.subTitle && <ExtendedText>{t(data.subTitle)}</ExtendedText>}
```
Comment (Reykjavik151) [inline app/components/Carousel/components/ImageTitleSubTitle/ImageTitleSubTitle.tsx]: the same as above

File: app/screens/Onboarding/Parents/CharmsIntroducing/CharmsIntroducing.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+export const CharmsIntroducingScreen: React.FC<ICharmsIntroducingScreenProps> =
+  ({ navigation }) => {
+    const { t } = useTranslation();
+
+    const [currentPosition, setCurrenPosition] = useState<number>(0);
+
+    return (
+      // TODO: remove noop to real function
+      <SafeAreaView style={[generalStyles.whFlex]}>
```
Comment (Reykjavik151) [inline app/screens/Onboarding/Parents/CharmsIntroducing/CharmsIntroducing.tsx]: array?

File: app/services/localization/languages/en.ts [hunk 11]
```diff
@@ -28,6 +29,25 @@ export const en = {
         subTitle:
           'I am so sorry that you have had to experience this. I am here to guide you and your child to find a way through this.',
       },
+      charms_introducing: {
+        carousel: {
```
Comment (Reykjavik151) [inline app/services/localization/languages/en.ts]: let's remove `carousel` from this screen translation
so from charms_introducing.carousel.first -> charms_introducing.first

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

**Review by Reykjavik151 — APPROVED** [review]

**Pull request merged**